substrate/frame/core-fellowship/src/lib.rs

```diff
@@ -578,6 +589,81 @@ pub mod pallet {
 				Self::deposit_event(e);
 			}
 		}
+
+		/// Ensure the correctness of the state of this pallet
+		/// The following expectations must always apply.
+		///
+		/// ## Expectations:
+		///
+		/// `ranked_member`:
+		/// * members with zero demotion period should not submit an evidence with a wish to retain,
+		///   subsequently no promotion
+		/// available for max_rank() members.
+		///
+		/// `unranked_member`:
+		/// * evidence should only be submitted with a `Wish:Promotion`.
+		#[cfg(any(feature = "try-runtime", test))]
+		pub fn do_try_state() -> Result<(), TryRuntimeError> {
+			// Check invariants for each member tracked by the pallet
+			let params = Params::<T, I>::get();
+
+			for who in Member::<T, I>::iter_keys() {
+				if let Some(_) = Member::<T, I>::get(who.clone()) {
+					let rank =
+						T::Members::rank_of(&who).ok_or("Member not found in rank registry")?;
+
+					if rank > T::Members::min_rank() {
+						Self::ranked_member(rank, &who, params.clone())?;
+					} else {
+						Self::unranked_member(&who)?;
+					}
+				}
+			}
+			Ok(())
+		}
+		#[cfg(any(feature = "try-runtime", test))]
+		fn ranked_member(
```

Comment:
```suggestion
		fn check_ranked_member(
```
Or something like this.